app/routes/_auth.signup.tsx

```diff
@@ -66,20 +93,75 @@ export default function Component() {
         <Card>
           <Form method="post" className="w-full flex flex-col items-center justify-center gap-5">
             <h1 className="text-4xl">Sign Up</h1>
-            <Input name="email" type="email" autoComplete="email" placeholder="Email" label="Email" required />
-            <Input name="name" autoComplete="username" type="text" placeholder="Name" label="Name" required />
-            <Input name="password" type="password" placeholder="Password" label="Password" required />
+
+            <Input
+              defaultValue={actionData?.fields?.name}
```

Comment:
Just for clarification. It seems that the Form component keeps input fields after validation. Should we use defaultValue?

Reply:
Yes, I think we should still keep `defaultValue` since if JavaScript fails to load for any reason, there will be a full page reload. In this case, setting the `defaultValue` will be a good fallback.

This is mentioned in the Remix App tutorial as well in the Mutations [section](https://remix.run/docs/en/main/tutorials/jokes#mutations): 

> You may also notice that I return the fields as well. This is so that the form can be re-rendered with the values from the server in the event that JavaScript fails to load for some reason. That's what the `defaultValue` stuff is all about as well.